src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -118,6 +118,21 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
   }
 
+  toPreviousPage() {
+    let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
+    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
+    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
+    caseEventData.event_token = this.eventTrigger.event_token;
+    caseEventData.ignore_warning = this.ignoreWarning;
+    caseEventData.event_data = this.editForm.value.data;
+
+    if (this.caseEdit.caseDetails) {
+      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
+    }
```

Comment:
this is duplicated of the code in `submit` method. We should removed this duplication.

Reply:
Mario  I am planning to create a generic method for this only.  It will be re-used  for submit() and  toPreviousPage() . Is it OK?

let pageFormFields = this.formValueService.filterCurrentPageFields(this.currentPage.case_fields, this.editForm.value);
    this.formValueService.sanitiseDynamicLists(this.currentPage.case_fields, pageFormFields);
    let caseEventData: CaseEventData = this.formValueService.sanitise(pageFormFields) as CaseEventData;
    caseEventData.event_token = this.eventTrigger.event_token;
    caseEventData.ignore_warning = this.ignoreWarning;
    caseEventData.event_data = this.editForm.value.data;

    if (this.caseEdit.caseDetails) {
      caseEventData.case_reference = this.caseEdit.caseDetails.case_id;
    }